### PR TITLE
rocksdb: add Env related API

### PR DESCRIPTION
**Pull request by choleraehyq**
PTAL @zhangjinpeng1987 @siddontang @andelf 
Signed-off-by: Cholerae Hu <huyingqian@pingcap.com>

### CODE REVIEW

File: src/rocksdb_options.rs [hunk 1]
```diff
@@ -949,6 +974,10 @@ impl Options {
             crocksdb_ffi::crocksdb_options_set_allow_concurrent_memtable_write(self.inner, v);
         }
     }
+
+    pub fn set_env(&mut self, env: &Env) {
```
Comment (siddontang) [inline src/rocksdb_options.rs]: I think you should keep the Env in the Options. Otherwise, the Env may be destroyed quickly when dropping Env.

Comment (choleraehyq) [inline src/rocksdb_options.rs]: We need to share Env among two or more db instance, but Options cannot be shared. We should keep Env's lifetime as the same to databases.

Comment (siddontang) [inline src/rocksdb_options.rs]: It is bad to let the users manage the lifetime of Env. We should only DB outside only.

File: librocksdb_sys/src/lib.rs [hunk 2]
```diff
@@ -216,6 +217,13 @@ pub enum DBTableProperty {
     CompressionName = 17,
 }
 
+#[repr(C)]
+pub enum Priority {
```
Comment (zhangjinpeng1987) [inline librocksdb_sys/src/lib.rs]: ThreadPoolPriority

File: src/rocksdb_options.rs [hunk 3]
```diff
@@ -294,6 +294,31 @@ impl Drop for CompactOptions {
...
+}
+
+impl Env {
+    pub fn new() -> Env {
+        unsafe { Env { inner: crocksdb_ffi::crocksdb_create_default_env() } }
+    }
+
+    pub fn set_background_threads(&mut self, n: Priority) {
+        unsafe {
+            crocksdb_ffi::crocksdb_env_set_background_threads(self.inner, n);
```
Comment (zhangjinpeng1987) [inline src/rocksdb_options.rs]: Miss number parameter

Comment (zhangjinpeng1987) [inline src/rocksdb_options.rs]: You should also use `crocksdb_env_set_high_priority_background_threads`.

**Comment by choleraehyq** [comment]
After discuss with jinpeng, we found out that thread pool is a singleton, we don't need to expose Env API. So close this PR.

**Comment by siddontang** [comment]
so, how do we share the same Env with multi rocksdbs?

**Comment by choleraehyq** [comment]
We do not need to do this. All rocksdb will share one thread pool implicitly.

**Pull request closed without merging**